### PR TITLE
Migrate to the new conversion generator - part1

### PR DESCRIPTION
**Pull request by wojtek-t**
This PR contains two commits:
- few more fixes to the generator
- migration of the pkg/api/v1 to use the new generator

The second commit is big, but I reviewed the changes and they contain:
- conversions between types that we didn't even generating conversion between
- changes in how we handle maps/pointers/slices - previously we were explicitly referencing fields, now we are using "shadowing in, out" to make the code more generic
- lack of auto-generated method for ReplicationControllerSpec (because these types are different (*int vs int for Replicas) and a preexisting conversion already exists

Most of issues in the first commit (e.g. adding references to "in" and "out" for slices/maps/points) were discovered by our tests. So I'm pretty confident that this change is correct now.

### CODE REVIEW

File: pkg/api/v1/types.go [hunk 1]
```diff
@@ -1816,6 +1816,7 @@ type ServiceSpec struct {
 	// API for compatibility until at least 8/20/2016.  It will be removed from
 	// any new API revisions.  If both deprecatedPublicIPs *and* externalIPs are
 	// set, deprecatedPublicIPs is used.
+	// +genconversion=false
```
Comment (lavalamp) [inline pkg/api/v1/types.go]: I think we've been adding these tags in a separate block so they don't show up in docs? e.g.:

```
// +genconversion=false

// actual comment for docs
MyVariable string
```

**Comment by lavalamp** [comment]
LGTM just one nit-- add the tag after you fix :)

File: pkg/api/v1/types.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/api/v1/types.go]: Actually - it depends :)
Basically, for generating client we use "second closest comment line", but e.g. for protobuf generation we are using the main comment, e.g.:
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/unversioned/time.go#L30

And for member fields we don't have SecondClosestComment lines yet.

I've added a TODO and will fix that in a separate PR (this is just a single place for now, so shouldn't cause big user problems :))

**Comment by wojtek-t** [comment]
Applying lgtm labed based on the above comment.

**Comment by wojtek-t** [comment]
Simple rebase - readding lgtm

**Comment by wojtek-t** [comment]
@k8s-bot test this please github issue: #IGNORE

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please github issue: #23645

**Comment by wojtek-t** [comment]
@k8s-bot unit test this please github issue; #IGNORE (test results not found)

**Comment by eparis** [comment]
I added the 'release-note' label so the bot I'm about to turn on won't get angry. If you do not believe this should be mentioned in the release notes please change the label to `release-note-none`

**Comment by wojtek-t** [comment]
To be honest - I think there is no need for release note.
@lavalamp - WDYT?

**Comment by lavalamp** [comment]
@wojtek-t agree

**Pull request merged**